Metrics api polishing

cmebarrow opened the pull request:
I found I was making a lot of comments about the API when reviewing your [PR](https://github.com/kaazing/metrics-demo/pull/10) so I thought it would save time if I just make the changes needed in another pull request. That way you can review the changes and merge them to your repository. Hopefully that will speed things up - it's a bit difficult to go back and forth with the 10 hour time difference.  I am being picky about the api. because we are expecting that customers will use the metrics demo projects, especially reader, as the starting point to build their own custom monitoring systems, so we want the code to be as streamlined and clear as possible, an example of how we think customers should write their code. 

I like the idea of having the org.kaazing.monitoring.reader.api package.The purpose of these  changes is to have that package be the only external API used by other projects (viewer and statsd), and have it contain only classes or interfaces necessary for that. The changes are only renaming, there is no change the logic of how things work. Here's a summary of the changes:
- Move MMFReaderBuilder, MetricsCollector, and MonitoringDataProcessor from reader api package to impl
- Add static wrap method to MMFReader interface
- Replase MMFReader methods getGatewayID and getGatewayCounters with getGateway (returning GatewayCounters)
- Rename GatewayCounters method getGatewayId to getId
- Change AgronaMonitoringDataProcessor to use MMFReader.wrap()
- Rename MMFReader to Metrics and improve the javadoc
- Move MonitoringDataProcessor back to api package and rename to MetricsFileProcessor
- Add newInstance method
- Rename MonitoringFolderAgrona to MonitoringFolder
- Rename MonitoringFolderAgronaImpl to MonitoringFolderImpl
- Rename related unit test classes

Commit:
- Move MMFReaderBuilder, MetricsCollector, and MonitoringDataProcessor from reader api package to impl.
- Add static wrap method to MMFReader interface.
- Replace MMFReader methods getGatewayID and getGatewayCounters with getGateway (returning GatewayCounters).
- Rename GatewayCounters method getGatewayId to getId.
- Change AgronaMonitoringDataProcessor to use MMFReader.wrap().

Commit:
- Rename MMFReader to Metrics and tweak the javadoc
- Move MonitoringDataProcessor back to api package, rename to MetricsFileProcessor
  and add newInstance method
- Rename MonitoringFolderAgrona to MonitoringFolder
- Rename MonitoringFolderAgronaImpl to MonitoringFolderImpl
- Rename related unit test classes

mgherghe commented on reader/src/main/java/org/kaazing/monitoring/reader/api/GatewayCounters.java:
```diff
@@ -24,14 +24,14 @@
      * @return String
      */
-    String getGatewayId();
+    String getId();
```
I think that renaming this method to getId, in the GatewayCounters' context, might make it seem like it would return the id of a GatewayCounters instance, and not the id of the gateway.

cmebarrow replied:
True. I did think about renaming GatewayCounters to Gateway (which which case getId would be ok). Feel free to do that if you want. But for now I will rename the method back to getGatewayId.

Commit:
Rename GatewayCounters.getId back to getGatewayId.

Commit:
Merge remote-tracking branch 'marina/develop' into metrics_api_polishing

Commit:
Fix compile error in MetricsCollectorAgrona (wrong import). Remove bin directories which were accidentally added.

cmebarrow commented:
@mgherghe I applied your review comment and removed the bin directories I accidentally added back so should be good for you to merge now (that way the changes will get into your forked repos and into your pull request). You'll still be able to make changes of course.

Commit:
Change reader/pom.xml to add failsave plugin so the robot tests actually get executed during build! Fix an issue in one robot test and add TODO comment to it.

cmebarrow commented:
@mgherghe I just added another commit to add failsafe plugin to the reader pom so the robot test actually gets executed when you build. I also made a minor fix to the second robot test) so it now passes) and added a TODO comment to check the actual expected counters are present and have the correct values. Please do the TODO after merging my PR.
